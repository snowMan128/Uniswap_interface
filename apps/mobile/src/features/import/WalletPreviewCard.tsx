import React from 'react'
import { Flex, Icons, Text, TouchableArea, Unicon } from 'ui/src'
import { iconSizes } from 'ui/src/theme'
import { NumberType } from 'utilities/src/format/types'
import { DisplayNameText } from 'wallet/src/components/accounts/DisplayNameText'
import { useLocalizationContext } from 'wallet/src/features/language/LocalizationContext'
import { useDisplayName } from 'wallet/src/features/wallet/hooks'
import { ElementNameType } from 'wallet/src/telemetry/constants'

interface Props {
  address: string
  selected: boolean
  balance?: number | null
  onSelect: (address: string) => void
  name?: ElementNameType
  testID?: string
  hideSelectionCircle?: boolean
}

export const ADDRESS_WRAPPER_HEIGHT = 36
const UNICON_SIZE = 32

export default function WalletPreviewCard({
  address,
  selected,
  balance,
  onSelect,
  hideSelectionCircle,
  ...rest
}: Props): JSX.Element {
  const displayName = useDisplayName(address, { showLocalName: true })
  const { convertFiatAmountFormatted } = useLocalizationContext()

  const balanceFormatted = convertFiatAmountFormatted(balance, NumberType.FiatTokenQuantity)

  return (
    <TouchableArea
      backgroundColor={selected ? '$surface1' : '$surface2'}
      borderColor={selected ? '$surface3' : '$surface2'}
      borderRadius="$rounded20"
      borderWidth={1}
      px="$spacing16"
      py="$spacing16"
      onPress={(): void => onSelect(address)}
      {...rest}>
      <Flex row alignItems="center" jc="space-between">
        <Flex row ai="center" gap="$spacing12" height={ADDRESS_WRAPPER_HEIGHT} jc="flex-start">
          <Unicon address={address} size={UNICON_SIZE} />
          <Flex ai="flex-start">
            <DisplayNameText displayName={displayName} textProps={{ variant: 'body1' }} />
            {balance ? (
              <Text color="$neutral2" variant="subheading2">
                {balanceFormatted}
              </Text>
            ) : null}
          </Flex>
        </Flex>
        {!hideSelectionCircle && selected && (
          <Icons.Check color="$accent1" size={iconSizes.icon20} />
        )}
      </Flex>
    </TouchableArea>
  )
}
