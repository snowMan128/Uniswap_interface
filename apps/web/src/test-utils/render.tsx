import { MockedProvider } from '@apollo/client/testing'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { queries } from '@testing-library/dom'
import { RenderOptions, render } from '@testing-library/react'
import { RenderHookOptions, WrapperComponent, renderHook } from '@testing-library/react-hooks'
import Web3Provider from 'components/Web3Provider'
import { AssetActivityProvider } from 'graphql/data/apollo/AssetActivityProvider'
import { TokenBalancesProvider } from 'graphql/data/apollo/TokenBalancesProvider'
import { BlockNumberContext } from 'lib/hooks/useBlockNumber'
import { PropsWithChildren, ReactElement, ReactNode } from 'react'
import { HelmetProvider } from 'react-helmet-async/lib/index'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import store from 'state'
import { ThemeProvider } from 'theme'
import { TamaguiProvider } from 'theme/tamaguiProvider'
import { UnitagUpdaterContextProvider } from 'uniswap/src/features/unitags/context'

const queryClient = new QueryClient()

const BLOCK_NUMBER_CONTEXT = { fastForward: () => {}, block: 1234, mainnetBlock: 1234 }
function MockedBlockNumberProvider({ children }: PropsWithChildren) {
  return <BlockNumberContext.Provider value={BLOCK_NUMBER_CONTEXT}>{children}</BlockNumberContext.Provider>
}

const WithProviders = ({ children }: { children?: ReactNode }) => {
  return (
    <HelmetProvider>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <BrowserRouter>
            <Web3Provider>
              <MockedProvider showWarnings={false}>
                <AssetActivityProvider>
                  <TokenBalancesProvider>
                    <MockedBlockNumberProvider>
                      <UnitagUpdaterContextProvider>
                        <ThemeProvider>
                          <TamaguiProvider>{children}</TamaguiProvider>
                        </ThemeProvider>
                      </UnitagUpdaterContextProvider>
                    </MockedBlockNumberProvider>
                  </TokenBalancesProvider>
                </AssetActivityProvider>
              </MockedProvider>
            </Web3Provider>
          </BrowserRouter>
        </QueryClientProvider>
      </Provider>
    </HelmetProvider>
  )
}

type CustomRenderOptions = Omit<RenderOptions, 'wrapper'>
const customRender = (ui: ReactElement, options?: CustomRenderOptions) => {
  return render<typeof queries>(ui, { ...options, wrapper: WithProviders })
}

type CustomRenderHookOptions<Props> = Omit<RenderHookOptions<Props>, 'wrapper'>
const customRenderHook = <Result, Props>(
  hook: (initialProps: Props) => Result,
  options?: CustomRenderHookOptions<Props>
) => {
  return renderHook(hook, { ...options, wrapper: WithProviders as WrapperComponent<Props> })
}

// Testing utils may export *.
// eslint-disable-next-line no-restricted-syntax
export * from '@testing-library/react'
export { customRender as render, customRenderHook as renderHook }
