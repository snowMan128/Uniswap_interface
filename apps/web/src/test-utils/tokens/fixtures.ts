import { QueryResult } from '@apollo/client'
import { Chain, Exact, TokenProjectQuery } from 'graphql/data/__generated__/types-and-hooks'

export const validTokenProjectResponse = {
  data: {
    token: {
      id: 'VG9rZW46RVRIRVJFVU1fMHhBMGI4Njk5MWM2MjE4YjM2YzFkMTlENGEyZTlFYjBjRTM2MDZlQjQ4',
      decimals: 6,
      name: 'USD Coin',
      chain: 'ETHEREUM',
      address: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
      symbol: 'USDC',
      standard: 'ERC20',
      project: {
        id: 'VG9rZW5Qcm9qZWN0OkVUSEVSRVVNXzB4YTBiODY5OTFjNjIxOGIzNmMxZDE5ZDRhMmU5ZWIwY2UzNjA2ZWI0OF9VU0RD',
        description:
          'USDC is a fully collateralized US dollar stablecoin. USDC is the bridge between dollars and trading on cryptocurrency exchanges. The technology behind CENTRE makes it possible to exchange value between people, businesses and financial institutions just like email between mail services and texts between SMS providers. We believe by removing artificial economic borders, we can create a more inclusive global economy.',
        homepageUrl: 'https://www.circle.com/en/usdc',
        twitterName: 'circle',
        logoUrl:
          'https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/logo.png',
        __typename: 'TokenProject',
      },
      __typename: 'Token',
    },
  },
} as unknown as QueryResult<TokenProjectQuery, Exact<{ chain: Chain; address?: string }>>
