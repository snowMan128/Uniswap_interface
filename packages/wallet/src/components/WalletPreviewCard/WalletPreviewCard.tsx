import { Flex, Icons, Text, TouchableArea, Unicon, UniconV2 } from 'ui/src'
import { iconSizes } from 'ui/src/theme'
import { NumberType } from 'utilities/src/format/types'
import { DisplayNameText } from 'wallet/src/components/accounts/DisplayNameText'
import { FEATURE_FLAGS } from 'wallet/src/features/experiments/constants'
import { useFeatureFlag } from 'wallet/src/features/experiments/hooks'
import { useLocalizationContext } from 'wallet/src/features/language/LocalizationContext'
import { useDisplayName } from 'wallet/src/features/wallet/hooks'
import { ElementNameType } from 'wallet/src/telemetry/constants'

interface Props {
  address: string
  selected: boolean
  balance?: number | null
  onSelect: (address: string) => void
  name?: ElementNameType
  testID?: string
  hideSelectionCircle?: boolean
}

export const ADDRESS_WRAPPER_HEIGHT = 36
const UNICON_SIZE = 32

export default function WalletPreviewCard({
  address,
  selected,
  balance,
  onSelect,
  hideSelectionCircle,
  ...rest
}: Props): JSX.Element {
  const isUniconsV2Enabled = useFeatureFlag(FEATURE_FLAGS.UniconsV2)

  const displayName = useDisplayName(address, { showLocalName: true })
  const { convertFiatAmountFormatted } = useLocalizationContext()

  const balanceFormatted = convertFiatAmountFormatted(balance, NumberType.FiatTokenQuantity)

  return (
    <TouchableArea
      backgroundColor={selected ? '$surface1' : '$surface2'}
      borderColor={selected ? '$surface3' : '$surface2'}
      borderRadius="$rounded20"
      borderWidth={1}
      px="$spacing16"
      py="$spacing16"
      onPress={(): void => onSelect(address)}
      {...rest}>
      <Flex row alignItems="center" justifyContent="space-between">
        <Flex
          row
          alignItems="center"
          gap="$spacing12"
          height={ADDRESS_WRAPPER_HEIGHT}
          justifyContent="flex-start">
          {isUniconsV2Enabled ? (
            <UniconV2 address={address} size={UNICON_SIZE} />
          ) : (
            <Unicon address={address} size={UNICON_SIZE} />
          )}
          <Flex alignItems="flex-start">
            <DisplayNameText displayName={displayName} textProps={{ variant: 'body1' }} />
            {balance ? (
              <Text color="$neutral2" variant="subheading2">
                {balanceFormatted}
              </Text>
            ) : null}
          </Flex>
        </Flex>
        {!hideSelectionCircle && selected && (
          <Icons.Check color="$accent1" size={iconSizes.icon20} />
        )}
      </Flex>
    </TouchableArea>
  )
}
