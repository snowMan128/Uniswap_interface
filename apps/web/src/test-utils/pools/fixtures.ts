import { BigNumber } from '@ethersproject/bignumber'
import { ChainId, Currency, WETH9 } from '@uniswap/sdk-core'
import { FeeAmount, Pool, Position } from '@uniswap/v3-sdk'
import { USDC_MAINNET } from 'constants/tokens'
import { Token as BEToken } from 'graphql/data/__generated__/types-and-hooks'
import { PoolData } from 'graphql/data/pools/usePoolData'
import { Token } from 'graphql/thegraph/__generated__/types-and-hooks'

export const validParams = { poolAddress: '0x88e6a0c2ddd26feeb64f039a2c41296fcb3f5640', chainName: 'ethereum' }

export const validPoolToken0 = {
  id: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
  symbol: 'USDC',
  name: 'USD Coin',
  decimals: '6',
  derivedETH: '0.0006240873011635544626425964678706127',
  __typename: 'Token',
} as Token
export const validBEPoolToken0 = {
  ...validPoolToken0,
  address: validPoolToken0.id,
  chain: 'ETHEREUM',
  decimals: 6,
} as BEToken

export const validUSDCCurrency = {
  isNative: false,
  isToken: true,
  name: 'USDCoin',
  address: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
  symbol: 'USDC',
  decimals: 6,
  chainId: 1,
  logoURI:
    'https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/ethereum/assets/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/logo.png',
  _checksummedAddress: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48',
  _tags: null,
  wrapped: validPoolToken0,
} as unknown as Currency

const validPoolToken1 = {
  id: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
  symbol: 'WETH',
  name: 'Wrapped Ether',
  decimals: '18',
  derivedETH: '1',
  __typename: 'Token',
} as Token
export const validBEPoolToken1 = {
  ...validPoolToken1,
  address: validPoolToken1.id,
  chain: 'ETHEREUM',
  decimals: 18,
} as BEToken

export const owner = '0xf5b6bb25f5beaea03dd014c6ef9fa9f3926bf36c'

const pool = new Pool(
  USDC_MAINNET,
  WETH9[ChainId.MAINNET],
  FeeAmount.MEDIUM,
  '1851127709498178402383049949138810',
  '7076437181775065414',
  201189
)

const position = new Position({
  pool,
  liquidity: 1341008833950736,
  tickLower: 200040,
  tickUpper: 202560,
})
const details = {
  nonce: BigNumber.from('0'),
  tokenId: BigNumber.from('0'),
  operator: '0x0',
  token0: USDC_MAINNET.address,
  token1: WETH9[ChainId.MAINNET].address,
  fee: FeeAmount.MEDIUM,
  tickLower: -100,
  tickUpper: 100,
  liquidity: BigNumber.from('9000'),
  feeGrowthInside0LastX128: BigNumber.from('0'),
  feeGrowthInside1LastX128: BigNumber.from('0'),
  tokensOwed0: BigNumber.from('0'),
  tokensOwed1: BigNumber.from('0'),
}
export const useMultiChainPositionsReturnValue = {
  positions: [
    {
      owner,
      chainId: ChainId.MAINNET,
      position,
      pool,
      details,
      inRange: true,
      closed: false,
    },
  ],
  loading: false,
}

export const usdcWethPoolAddress = '0x88e6a0c2ddd26feeb64f039a2c41296fcb3f5640'

export const validPoolDataResponse = {
  data: {
    feeTier: 500,
    token0: validBEPoolToken0,
    token1: validBEPoolToken1,
    token0Price: 1605.481,
    token1Price: 0.000622,
    txCount: 5456494,
    address: usdcWethPoolAddress,
    volumeUSD24HChange: -17.753809465717136,
    volumeUSD24H: 194273059.265625,
    tvlUSD: 223166198.4690675,
    tvlUSDChange: -0.3657085465786977,
    tvlToken0: 90930713.7356909,
    tvlToken1: 82526.48678530742,
  } as PoolData,
  loading: false,
  error: false,
}
