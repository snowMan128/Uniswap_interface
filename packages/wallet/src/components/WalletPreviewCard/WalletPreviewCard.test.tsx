import { SAMPLE_SEED_ADDRESS_1 } from 'wallet/src/test/fixtures'
import { render } from 'wallet/src/test/test-utils'
import WalletPreviewCard from './WalletPreviewCard'

it('renders wallet preview card', () => {
  const tree = render(
    <WalletPreviewCard selected address={SAMPLE_SEED_ADDRESS_1} onSelect={(): null => null} />
  )
  expect(tree).toMatchSnapshot()
})
