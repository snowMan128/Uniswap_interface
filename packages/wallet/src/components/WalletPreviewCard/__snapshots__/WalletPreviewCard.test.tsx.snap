// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders wallet preview card 1`] = `
<View
  cancelable={true}
  focusable={true}
  hitSlop={[Function]}
  minPressDuration={0}
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#2222220D",
      "borderBottomLeftRadius": 20,
      "borderBottomRightRadius": 20,
      "borderBottomWidth": 1,
      "borderLeftColor": "#2222220D",
      "borderLeftWidth": 1,
      "borderRightColor": "#2222220D",
      "borderRightWidth": 1,
      "borderStyle": "solid",
      "borderTopColor": "#2222220D",
      "borderTopLeftRadius": 20,
      "borderTopRightRadius": 20,
      "borderTopWidth": 1,
      "flexDirection": "column",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "transform": [
        {
          "scale": 1,
        },
      ],
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 12,
          "height": 36,
          "justifyContent": "flex-start",
        }
      }
    >
      <div
        style={
          {
            "height": 32,
            "position": "relative",
            "width": 32,
          }
        }
      >
        <div
          style={
            {
              "height": 32,
              "overflow": "visible",
              "position": "absolute",
              "width": 32,
            }
          }
        >
          <svg
            viewBox="0 0 32 32"
          >
            <defs>
              <defs>
                <mask
                  id="container-mask0x82D56A352367453f74FC0dC7B071b311da373Fa632"
                >
                  <rect
                    fill="white"
                    height="100%"
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <g
                    transform="scale(0.8888888888888888) 
 translate(0, 0)"
                  >
                    <path
                      clipRule="evenodd"
                      d="M31 33V16C31 8.8203 25.1797 3 18 3C10.8203 3 5 8.8203 5 16V33H31ZM18 0C9.16344 0 2 7.16344 2 16V35.807C2 35.9136 2.08642 36 2.19303 36H33.807C33.9136 36 34 35.9136 34 35.807V16C34 7.16344 26.8366 0 18 0Z"
                      fill="black"
                      fillRule="evenodd"
                    />
                  </g>
                </mask>
                <mask
                  id="shape-mask0x82D56A352367453f74FC0dC7B071b311da373Fa632"
                >
                  <rect
                    fill="white"
                    height="100%"
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <g
                    transform="scale(0.8888888888888888) 
 translate(10, 10)"
                  >
                    <path
                      d="M8 8C8 8 13.1678 6.70089 14.2197 5.649C15.2716 4.59712 15.2583 2.87836 14.19 1.81004C13.1216 0.741719 11.4029 0.728395 10.351 1.78028C9.29911 2.83216 8 8 8 8Z"
                      fill="black"
                    />
                    <path
                      d="M8 8C8 8 13.1678 9.29911 14.2197 10.351C15.2716 11.4029 15.2583 13.1216 14.19 14.19C13.1216 15.2583 11.4029 15.2716 10.351 14.2197C9.29911 13.1678 8 8 8 8Z"
                      fill="black"
                    />
                    <path
                      d="M1.78028 10.351C2.83216 9.29911 8 8 8 8C8 8 6.70089 13.1678 5.649 14.2197C4.59712 15.2716 2.87836 15.2583 1.81004 14.19C0.741719 13.1216 0.728395 11.4029 1.78028 10.351Z"
                      fill="black"
                    />
                    <path
                      d="M8 8C8 8 2.83216 6.70089 1.78028 5.649C0.728395 4.59712 0.741719 2.87836 1.81004 1.81004C2.87836 0.741719 4.59712 0.728395 5.649 1.78028C6.70089 2.83216 8 8 8 8Z"
                      fill="black"
                    />
                  </g>
                </mask>
                <mask
                  id="mask0x82D56A352367453f74FC0dC7B071b311da373Fa632"
                >
                  <g
                    fill="white"
                  >
                    <g
                      mask="url(#shape-mask0x82D56A352367453f74FC0dC7B071b311da373Fa632)"
                    >
                      <g
                        transform="scale(0.8888888888888888)"
                      >
                        <path
                          clipRule="evenodd"
                          d="M31 33V16C31 8.8203 25.1797 3 18 3C10.8203 3 5 8.8203 5 16V33H31ZM18 0C9.16344 0 2 7.16344 2 16V35.807C2 35.9136 2.08642 36 2.19303 36H33.807C33.9136 36 34 35.9136 34 35.807V16C34 7.16344 26.8366 0 18 0Z"
                          fillRule="evenodd"
                        />
                      </g>
                    </g>
                    <g
                      mask="url(#container-mask0x82D56A352367453f74FC0dC7B071b311da373Fa632)"
                    >
                      <g
                        transform="scale(0.8888888888888888)
        translate(10, 10)"
                      >
                        <path
                          d="M8 8C8 8 13.1678 6.70089 14.2197 5.649C15.2716 4.59712 15.2583 2.87836 14.19 1.81004C13.1216 0.741719 11.4029 0.728395 10.351 1.78028C9.29911 2.83216 8 8 8 8Z"
                        />
                        <path
                          d="M8 8C8 8 13.1678 9.29911 14.2197 10.351C15.2716 11.4029 15.2583 13.1216 14.19 14.19C13.1216 15.2583 11.4029 15.2716 10.351 14.2197C9.29911 13.1678 8 8 8 8Z"
                        />
                        <path
                          d="M1.78028 10.351C2.83216 9.29911 8 8 8 8C8 8 6.70089 13.1678 5.649 14.2197C4.59712 15.2716 2.87836 15.2583 1.81004 14.19C0.741719 13.1216 0.728395 11.4029 1.78028 10.351Z"
                        />
                        <path
                          d="M8 8C8 8 2.83216 6.70089 1.78028 5.649C0.728395 4.59712 0.741719 2.87836 1.81004 1.81004C2.87836 0.741719 4.59712 0.728395 5.649 1.78028C6.70089 2.83216 8 8 8 8Z"
                        />
                      </g>
                    </g>
                  </g>
                </mask>
              </defs>
              <linearGradient
                id="gradient0x82D56A352367453f74FC0dC7B071b311da373Fa632"
              >
                <stop
                  offset="0%"
                  stopColor="#6100FF"
                />
                <stop
                  offset="100%"
                  stopColor="#76D191"
                />
              </linearGradient>
              <filter
                height="200%"
                id="blur0x82D56A352367453f74FC0dC7B071b311da373Fa632"
                width="200%"
                x="-50%"
                y="-50%"
              >
                <feGaussianBlur
                  in="SourceGraphic"
                  stdDeviation={10.666666666666666}
                />
              </filter>
            </defs>
            <g
              mask="url(#mask0x82D56A352367453f74FC0dC7B071b311da373Fa632)"
            >
              <rect
                fill="url(#gradient0x82D56A352367453f74FC0dC7B071b311da373Fa632)"
                height="100%"
                width="100%"
                x="0"
                y="0"
              />
              <rect
                fill="black"
                height="100%"
                opacity={0.08}
                width="100%"
                x="0"
                y="0"
              />
              <ellipse
                cx={16}
                cy={0}
                fill="#D3EBA3"
                filter="url(#blur0x82D56A352367453f74FC0dC7B071b311da373Fa632)"
                rx={16}
                ry={16}
              />
            </g>
          </svg>
        </div>
      </div>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={true}
            maxFontSizeMultiplier={1.4}
            numberOfLines={1}
            style={
              {
                "color": "#222222",
                "flexShrink": 1,
                "fontFamily": "Basel-Book",
                "fontSize": 19,
                "lineHeight": 24,
              }
            }
            suppressHighlighting={true}
          >
            0xâ€‹82D5...3Fa6
          </Text>
        </View>
      </View>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      strokeWidth={8}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "color": "#FC72FF",
            "height": 20,
            "width": 20,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      tintColor="#FC72FF"
      vbHeight={16}
      vbWidth={16}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "strokeWidth",
          ]
        }
        strokeWidth="8"
      >
        <RNSVGPath
          d="M13.3334 4L6.00008 11.3333L2.66675 8"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
